d2bs/kolbot/libs/bots/MFHelper.js

```diff
@@ -157,7 +157,7 @@ MainLoop:
 					print("ÿc4MFHelperÿc0: Clear Cows");
 
 					for (i = 0; i < 5; i += 1) {
-						if (Town.goToTown(1) && Pather.usePortal(39)) {
+						if (Town.goToTown(1) && Pather.getPortal(39) && Pather.usePortal(39)) {
```

Comment:
this already will retry 5 times, so I don't think adding the getPortal helps

Reply:
yes, it retries 5 times, but if there is no portal then usePortal() will take *at least* 10 seconds looking for a portal. if we use getPortal() before then we know almost instantly if we should usePortal() or not.

Reply:
The thing is, if the delay in `usePortal()` is intentional to recover from desync, your code addition in MFHelper will break that approach by skipping the delay...

If it takes at least 10 seconds it looks like you have a really bad ping. Anyways, I think it depends on what you want to achieve. IMHO the timeout counter should instead be an optional argument for the `usePortal()` function:
```js
usePortal: function (targetArea, owner, unit, retries=10)
```
because I think there are scripts that depend on the portal and might be willing to wait even longer for a portal to exist.
Then you could use:
```js
if (Town.goToTown(1) && Pather.usePortal(39,null,null,0)) {
```
To return after the first delay passed.. that would allow you to pull out the resync handling from Pather.js without basically rewriting it.